uvq_pytorch/utils/distortionnet.py

```diff
@@ -0,0 +1,356 @@
+            nn.Sigmoid(),
+        )
+
+    def forward(self, x):
+        x = self.features(x)
+        features = self.avgpool(x)
+        label_probs = self.classifier(features)
+        return features, label_probs
+
+
+class DistortionNetInference:
+    def __init__(
+        self,
+        model_path=MODEL_PATH,
+        eval_mode=True,
+        pretrained=True,
+        video_height=VIDEO_HEIGHT,
+        video_width=VIDEO_WIDTH,
+        video_channels=VIDEO_CHANNELS,
+        patch_height=PATCH_HEIGHT,
+        patch_width=PATCH_WIDTH,
+        depth=PATCH_DEPTH,
+        feature_channels=DIM_CHANNEL_FEATURE,
+        feature_height=DIM_HEIGHT_FEATURE,
+        feature_width=DIM_WIDTH_FEATURE,
+        label_dim=OUTPUT_LABEL_DIM,
+    ):
+        self.model = DistortionNet()
+        if pretrained:
+            self.load_state_dict(model_path)
+        if eval_mode:
+            self.model.eval()
+        self.num_patches_x = int(video_width / patch_width)
+        self.num_patches_y = int(video_height / patch_height)
+        self.feature_channels = feature_channels
+        self.feature_height = feature_height
+        self.feature_width = feature_width
+        self.patch_width = patch_width
+        self.patch_height = patch_height
+        self.video_width = video_width
+        self.video_height = video_height
+        self.video_channels = video_channels
+        self.depth = depth
+        self.patch_feature_height = int(feature_height / self.num_patches_y)
+        self.patch_feature_width = int(feature_width / self.num_patches_x)
+        self.label_dim = label_dim
+
+    def load_state_dict(self, model_path) -> nn.Module:
+        model = torch.load(model_path)
+        self.model.load_state_dict(model)
+        return model
+
+    def predict(self, frame):
+        with torch.no_grad():
+            _, label_probs = self.model(Tensor(frame))
+        return label_probs.detach().numpy()
+
+    def predict_and_get_features(self, frame) -> tuple[np.ndarray, np.ndarray]:
+        with torch.no_grad():
+            features, label_probs = self.model(torch.Tensor(frame))
+        return (
+            features.detach().numpy(),
+            label_probs.detach().numpy(),
+        )
+
+    def get_labels_and_features_for_all_frames(
+        self,
+        video,
+    ):
+        # TODO: allow converting video to a batch of patches and running batch prediction instead of sliding window for loops
+        label = np.ndarray(
+            (video.shape[0], self.num_patches_y, self.num_patches_x, self.label_dim),
+            np.float32,
+        )
+        feature = np.ndarray(
+            (
+                video.shape[0],
+                self.feature_height,
+                self.feature_width,
+                self.feature_channels,
+            ),
+            np.float32,
+        )
+
+        if self.depth == 1:
+            # TODO: this is for distortionnet; for compressionnet keep only the "else" part.
```

Comment:
Maybe we can remove this comment and also the "else" part.

Maybe add an assertion somewhere to make sure depth is always 1 for now.